src/pybind/mgr/cephadm/module.py

```diff
@@ -2216,7 +2217,7 @@ def _calc_daemon_deps(self,
             except Exception:
                 pass
             deps = sorted([self.get_mgr_ip(), server_port, root_cert,
-                           str(self.get_module_option('device_enhanced_scan'))])
```

Comment:
When setting up the agent config originally as well
https://github.com/ceph/ceph/blob/master/src/pybind/mgr/cephadm/services/cephadmservice.py#L1038